src/backend/distributed/metadata/node_metadata.c

```diff
@@ -757,6 +757,14 @@ master_update_node(PG_FUNCTION_ARGS)
 		if (force)
 		{
 			handle = StartLockAcquireHelperBackgroundWorker(MyProcPid, lock_cooldown);
```

Comment:
It seems like we want to ignore the error that can happen in `StartLockAcquireHelperBackgroundWorker`. We changed one place where it can error but it might error in some other places as well. Wouldn't that make more sense to ignore errors there as:

```c
		PG_TRY();
		{
			handle = StartLockAcquireHelperBackgroundWorker(MyProcPid, lock_cooldown);
		}
			PG_CATCH();
		{
             ereport ( NOTICE, errmsg( "cannot start the background worker to kill conflicting locks since no slot is available, will switch back to normal mode from force");
             // set lock timeout here
		}
		PG_END_TRY();
```

What do you think @marcocitus?

That way we wouldn't need to return `NULL` in `StartLockAcquireHelperBackgroundWorker`.

Reply:
We want to ignore it specifically in master_update_node, not elsewhere.

Reply:
In general, PG_CATCH blocks that continue are not super safe because we might skip over important clean-up logic. What are the other errors?

Reply:
Currently there isn't any other error on our side I think. But there might be some errors on postgres side. For example:

StartLockAcquireHelperBackgroundWorker -> RegisterDynamicBackgroundWorker ->  LWLockAcquire. And there are errors in `LWLockAcquire`. Though I am not sure if we want to ignore those errors. I guess this might fall into "not super safe" category to continue.

Reply:
I think there's no reason to do that. In this PR we are addressing a well-understood runtime error in a very safe fashion.